src/base/object.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "quaternion.h"
+#include "vector.h"
+
+namespace gm {
+class SceneObject {
+ public:
+  SceneObject(const Vector3f &location = Vector3f(),
+              const Quaternionf &rotation = Quaternionf(),
+              const Vector3f &scale = Vector3f(1.f, 1.f, 1.f),
+              const std::string &name = "");
+
+  Vector3f get_location();
+  std::string name;
+  Vector3f location;
+  Quaternionf rotation;
+  Vector3f scale;
+  std::shared_ptr<SceneObject> parent;
+  std::vector<std::shared_ptr<SceneObject>> children;
+  virtual bool isMesh() { return false; }
```

Comment:
I am not sure using virtual functions like this is a good idea. It should be obvious when a scene object is a mesh, camera, and so on through inheritance. I don't see where these functions are called at the moment. What is the purpose of having them here?

Reply:
Fixed in 3b10693